### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by tobiashort** [dannyfiresnake/xl#4]
The locker breaks when entering too many characters.
I played around a bit and I think after 512 characters the locker breaks with a segmentation fault.

**Comment by uglywizkid** [comment]
Looks like a bug in crypt.h
```
#include <crypt.h>
#include <stdio.h>

int main() {
	char* enc = crypt("aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "$2a$10$eqCElfVfl4PYBp6dLRutnOfu0enMfq53cZR5IfPykJrDEpsMFRRz.");
	printf("%s\n", enc);
}
```

**Comment by uglywizkid** [comment]
Filed also a bug here:
https://sourceware.org/bugzilla/show_bug.cgi?id=27470

**Comment by dannydulai** [comment]
fixed in 93cff5f91d470479ba8b6825e6de84973e882cae
